x/executionlayer/handler.go

```diff
@@ -177,31 +185,41 @@ func handlerMsgUnBond(ctx sdk.Context, k ExecutionLayerKeeper, msg types.MsgUnBo
 }
 
 func execute(ctx sdk.Context, k ExecutionLayerKeeper, msg types.MsgExecute) (bool, string) {
-
+	proxyContractHash := k.GetProxyContractHash(ctx)
```

Comment:
Even though we can pay with stored_hash, shouldn't we also support payment with a wasm contract based on the cli flag?

Reply:
@caramis  doing an Execute cleanup, and once that part is decided probably expect the whole thing to change. After that, Let's change then.

Reply:
:ok_hand: